include/utils/system_reference_blas.hpp

```diff
@@ -79,6 +72,27 @@ CBLAS_DIAG c_diag(char x) {
       return CblasNonUnit;
   }
 }
+
+// i*amin is an extension, provide an implementation
+template<typename scalar_t>
+inline int iamin(const int N, const scalar_t *X, const int incX)
+{
+  int best = 0;
+  for(int i = 1; i < N; i += incX) {
+    if(std::abs(X[i]) < std::abs(X[best])) {
+      best = i;
```

Comment:
This should return i / incX

Reply:
Thanks for catching that @RossBrunton , fixed

Reply:
lgtm